src/WebsocketServer.js

```diff
@@ -65,24 +64,38 @@ module.exports = class WebsocketServer extends events.EventEmitter {
     }
 
     handlePublishRequest(connection, request) {
-        // Check that the payload is a string
-        if (typeof request.content !== 'string') {
-            connection.send(new Protocol.ErrorResponse('Message must be stringified JSON!'))
-            return
+        let streamId
+        if (request.version === 0) {
+            /* eslint-disable prefer-destructuring */
+            streamId = request.streamId
+            /* eslint-enable prefer-destructuring */
+        } else if (request.version === 1) {
+            streamId = request.streamMessage.getStreamId()
+        } else {
+            throw new Error(`Unrecognized version: ${request.version}`)
         }
-
-        this.streamFetcher.authenticate(request.streamId, request.apiKey, request.sessionToken, 'write')
-            .then((stream) => this.publisher.publish(
-                stream,
-                request.timestamp,
-                undefined, // ttl, read from stream when available
-                StreamrBinaryMessage.CONTENT_TYPE_JSON,
-                request.content,
-                request.partitionKey,
-                request.signatureType,
-                request.publisherAddress,
-                request.signature,
-            ))
+        this.streamFetcher.authenticate(streamId, request.apiKey, request.sessionToken, 'write')
+            .then((stream) => {
+                let streamMessage
+                if (request.version === 0) {
```

Comment:
This one may be harder to abstract, because in v0 you have the `partitionKey`, and in `v1` the partition is explicitly defined in the message. There's no common denominator, no common interface to be used. Options basically:
- Leave the `if` until we can drop v0
- Drop support for `partitionKey` from v0 protocol (just ignore it)